src/Agent.Service/Windows/AgentListener.cs

```diff
@@ -0,0 +1,326 @@
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace AgentService
+{
+    public enum ExecutionMode
+    {
+        Process,
+        Service
+    }
+
+    public class AgentListener
+    {
+        private ExecutionMode _currentExecutionMode;
+        private object ServiceLock { get; set; }
+        private Process _listenerProcess;
+        private bool Stopping { get; set; }
+        private bool _restart = false;
+        private const int CTRL_C_EVENT = 0;
+
+        public AgentListener(ExecutionMode mode)
+        {
+            _currentExecutionMode = mode;
```

Comment:
>_currentExecutionMode [](start = 12, length = 21)

we r not doing anything with mode here. what else is different in this implementation ?